griptape/tools/shopify_client/tool.py

```diff
@@ -0,0 +1,66 @@
+from textwrap import dedent
+from typing import Optional
+from urllib.parse import urljoin
+import schema
+from schema import Schema, Literal
+from attr import define, field
+from griptape.tools import BaseTool
+from griptape.utils.decorators import activity
+from griptape.artifacts import BaseArtifact, TextArtifact, ErrorArtifact
+
+
+@define
+class ShopifyClient(BaseTool):
+    storename: str = field(kw_only=True)
+    storefront_access_token: str = field(kw_only=True)
+    shopify_endpoint: str = field(default="myshopify.com/api/2023-07/graphql.json")
+
+    @activity(
+        config={
+            "description": "Can be used to search for shopify products",
+            "schema": Schema(
+                {
+                    Literal("first", description="How many products to retreive."): str,
+                }
+            ),
+        }
+    )
+    def search(self, params: dict) -> BaseArtifact:
+        from requests import post, exceptions
+
+        first = params.get("first", 10)
```

Comment:
I think we can set the default in the schema. But also, we should move this into one of the class properties, so the user can specify this number.

Reply:
@vasinov do you have an example of specifying a default value in a tool schema? I can do it via:
```
Optional(
  Literal(
      "product_count",
      description="How many products to retrieve.",
  ),
  default=10,
): str,
```
But the default value is not passed into the activity when the LLM calls it.

Regarding the class property, isn't this field kind of specific to this single API call?